src/raft.rs

```diff
@@ -1425,7 +1428,15 @@ impl<T: Storage> Raft<T> {
                         new_message(m.from, vote_resp_msg_type(m.get_msg_type()), None);
                     to_send.reject = true;
                     to_send.term = self.term;
+                    let (commit, commit_term) = self.raft_log.commit_info();
+                    to_send.commit = commit;
+                    to_send.commit_term = commit_term;
                     self.r.send(to_send, &mut self.msgs);
+                    // It should be safe to commit log by vote in all conditions as long as it's
+                    // not leader. Here use a stricter condition for best safety.
+                    if self.leader_id == INVALID_ID {
```

Comment:
Looks like only `(pre-)candidate`'s `leader_id` will be `INVALID_ID`, so we can't forward a follower's commit index by vote?

Reply:
When a peer's term is increased, its `leader_id` will be reset. So at the beginning of every term, `leader_id` of all followers and candidates should be `INVALID_ID`.

Reply:
Follower won't increase its term for `MsgRequestPreVote`

Reply:
It doesn't have to. It will increase its term when ticks over election timeout; otherwise there is already a leader, no need to advance commit index by messages from others.

Reply:
If the follower thinks itself is a learner, it won't increase its term and start campaign, and the issue still exists.